src/EditorFeatures/Core/Implementation/Suggestions/FixAllSuggestedAction.cs

```diff
@@ -23,16 +25,50 @@ internal sealed partial class FixAllSuggestedAction : SuggestedAction, ITelemetr
     {
         private readonly Diagnostic _fixedDiagnostic;
 
+        /// <summary>
+        /// The original code-action that we are a fix-all for.  i.e. _originalCodeAction
+        /// would be something like "use 'var' instead of 'int'", this suggestion action
+        /// and our <see cref="SuggestedAction.CodeAction"/> is the actual action that 
+        /// will perform the fix in the appropriate document/project/solution scope.
+        /// </summary>
+        private readonly CodeAction _originalCodeAction;
+        private readonly FixAllState _fixAllState;
+
         internal FixAllSuggestedAction(
             SuggestedActionsSourceProvider sourceProvider,
             Workspace workspace,
             ITextBuffer subjectBuffer,
             FixAllState fixAllState,
-            Diagnostic originalFixedDiagnostic)
+            Diagnostic originalFixedDiagnostic,
+            CodeAction originalCodeAction)
             : base(sourceProvider, workspace, subjectBuffer,
                    fixAllState.FixAllProvider, new FixAllCodeAction(fixAllState))
         {
             _fixedDiagnostic = originalFixedDiagnostic;
+            _originalCodeAction = originalCodeAction;
+            _fixAllState = fixAllState;
+        }
+
+        public override bool TryGetTelemetryId(out Guid telemetryId)
+        {
+            // We get the telemetry id for the original code action we are fixing,
+            // not the special 'FixAllCodeAction'.  that is the .CodeAction this
+            // SuggestedAction is pointing at.
+            var prefix = GetTelemetryPrefix(_originalCodeAction);
+            var scope = GetTelemetryScope();
+            telemetryId = new Guid(prefix, scope, 0, 0, 0, 0, 0, 0, 0, 0, 0);
```

Comment:
>Guid [](start = 30, length = 4)

Its unclear which Guid constructor we are calling here, but I think scope should be a short instead of a byte correct?  Why pass in a byte if its just going to be converted to a short?